selfdrive/boardd/panda.cc

```diff
@@ -0,0 +1,263 @@
+#include <stdexcept>
+#include <cassert>
+#include <iostream>
+
+#include "common/swaglog.h"
+
+#include "panda.h"
+
+Panda::Panda(){
+  int err;
+
+  err = pthread_mutex_init(&usb_lock, NULL);
+  if (err != 0) { goto fail; }
+
+  // init libusb
+  err = libusb_init(&ctx);
+  if (err != 0) { goto fail; }
+
+#if LIBUSB_API_VERSION >= 0x01000106
+  libusb_set_option(ctx, LIBUSB_OPTION_LOG_LEVEL, LIBUSB_LOG_LEVEL_INFO);
+#else
+  libusb_set_debug(ctx, 3);
+#endif
+
+  dev_handle = libusb_open_device_with_vid_pid(ctx, 0xbbaa, 0xddcc);
+  if (dev_handle == NULL) { goto fail; }
+
+  err = libusb_set_configuration(dev_handle, 1);
+  if (err != 0) { goto fail; }
+
+  err = libusb_claim_interface(dev_handle, 0);
+  if (err != 0) { goto fail; }
+
+  hw_type = get_hw_type();
+  is_pigeon =
+    (hw_type == cereal::HealthData::HwType::GREY_PANDA) ||
```

Comment:
think it makes sense to add a USB command for things like this, but in this instance, everything should be a pigeon now.

Reply:
You mean change the logic? So if not WHITE, PEDAL, UNKOWN, it's pigeon?

Reply:
We can do this later, but ideally there's a USB command to query board features. there's lists like this in a few different places in openpilot that should just be part of a panda abstraction